### PR TITLE
Igonchar/tcomp 317 converters

### PR DESCRIPTION
**Pull request by gonchar-ivan**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

DiIncomingSchemaEnforcer determines conversion strategy from scratch each time it receives some data value

**What is the new behavior?**

Conversions strategies (converters) are computed at the beginning of DiIncomingSchemaEnforcer usage.
Then these strategies are reused

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Related to https://jira.talendforge.org/browse/TCOMP-317

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/97/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/97/)

File: daikon/src/main/java/org/talend/daikon/di/converter/RegularConverter.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ * <li>long</li>
+ * <li>float</li>
+ * <li>double</li>
+ * <li>String</li>
+ * <li>Logical Time</li>
+ * <li>...</li>
+ * </ol>
+ * 
+ */
+public class RegularConverter extends AbstractAvroConverter<Object, Object> {
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/di/converter/RegularConverter.java]: the naming here is strange, may be we should find something that means no convertion is done.
something like PassthroughConverter or IdentityConverter.

Comment (gonchar-ivan) [inline daikon/src/main/java/org/talend/daikon/di/converter/RegularConverter.java]: Fully agree with you. 
I will think more. Thanks for suggestions

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/102/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/102/)

**Comment by gonchar-ivan** [comment]
I made rebase on latest master.
Rename RegularConverter to IdentityConverter by analogy to identity function in math
https://en.wikipedia.org/wiki/Identity_function
Fix unused import 
Add TypeConverter class

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/103/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/103/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/104/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/104/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/105/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/105/)

**Pull request merged**